File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/NewsDetailsModule.java
package com.androidacademy.msk.exerciseproject.di.module;

import android.support.annotation.NonNull;

import com.androidacademy.msk.exerciseproject.data.database.dao.NewsDao;
import com.androidacademy.msk.exerciseproject.di.scope.FragmentScope;
import com.androidacademy.msk.exerciseproject.screen.news_details.NewsDetailsPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class NewsDetailsModule {

    private final int currentItemId;

    public NewsDetailsModule(int currentItemId) {
        this.currentItemId = currentItemId;
    }

    @Provides
    @FragmentScope
    @NonNull
    public NewsDetailsPresenter providePresenter(NewsDao dao) {
        return new NewsDetailsPresenter(dao, currentItemId);
    }
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/NewsDetailsModule.kt
package com.androidacademy.msk.exerciseproject.di.module

import com.androidacademy.msk.exerciseproject.di.model.NewsId
import com.androidacademy.msk.exerciseproject.screen.newsdetails.NewsDetailsPresenter
import toothpick.config.Module
import javax.inject.Inject


class NewsDetailsModule @Inject constructor(newsId: NewsId) : Module() {
    init {
        bind(NewsId::class.java).toInstance(newsId)
        bind(NewsDetailsPresenter::class.java).singletonInScope()
    }
}